aea/crypto/fetchai_base.py

```diff
@@ -0,0 +1,117 @@
+
+from typing import Optional
+import logging
+from fetchai.ledger.crypto import Entity, Identity, Address  # type: ignore
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class FetchCryptoError(Exception):
+    """Exception to be thrown when cryptographic signatures don't match!."""
+
+
+class FetchCrypto(object):
+    """Class wrapping the Entity Generation from Fetch.AI ledger."""
+
+    def __init__(self, private_key_path: Optional[str] = None):
+        """Instantiate a crypto object."""
+        self._entity = self._generate_private_key() if private_key_path is None else self._load_private_key_from_path(private_key_path)
+        self._public_key_obj = self._entity.public_key
+        self._public_key_bytes = self._entity.public_key_bytes
+        self._public_key_hex = self._entity.public_key_hex
+        self._display_address = Address(Identity.from_hex(self._public_key_hex))
+        self._private_key = self._entity.private_key
+        self._private_key_hex = self._entity.private_key_hex
+        self._private_key_bytes = self._entity.private_key_bytes
+
+    @property
+    def public_key(self) -> str:
+        """
+        Return a public key in hex format.
+
+        :return: a public key string in hex format
+        """
+        return self._public_key_hex
+
+    @property
+    def display_address(self) -> str:
+        """
+        Return the display_address for the key pair.
+
+        :return: a display_address str
+        """
+        return str(self._display_address)
+
+    @staticmethod
+    def get_address_from_public_key(public_key: str) -> Address:
+        """
+        Get the address from the public key.
+
+        :return: str
+        """
+        identity = Identity.from_hex(public_key)
+        return Address(identity)
+
+    def sign_transaction(self, message: bytes) -> bytes:
+        """
+        Sing a transaction to send it to the ledger.
+
+        :param message:
+        :return: Signed message in bytes
+        """
+        signature = self._entity.sign(message)
+        return signature
+
+    def _load_private_key_from_path(self, file_name) -> Entity:
+        """
+        Load a private key in hex format from a file.
+
+        :param path: the path to the hex file.
+
+        :return: the Entity.
+        """
+        path = Path(file_name)
+        print(path)
+        try:
+            if path.is_file():
+                with open(path, "r") as key:
+                    data = key.read()
+                    print(data)
+                    entity = Entity.from_hex(data)
+
+            else:
+                entity = self._generate_private_key()
+
+            return entity
+        except IOError as e:
+            logger.exception(str(e))
+
+    def _generate_private_key(self) -> Entity:
+        path = Path("pk.txt")
+        print(path)
+        entity = Entity()
+        with open(path, "w+") as file:
```

Comment:
What if a private key file already exists?

Reply:
You just load it and you create the Entity.

Reply:
Yes, but your code will overwrite it. So basically, what I am suggesting with this question is that you need to adjust the code to check whether a private key exists, if so load it, otherwise, generate it.